### DISCUSSION TITLE
How do I get test progress or any console output during execution with dotnet test?

### DISCUSSION THREAD

**Discussion by turbobobbytraykov** [microsoft/testfx#3967] [Q&A] [answered]
I'm using the latest stable versions of Microsoft.NET.Test.Sdk, MSTest.TestFramework and MSTest.TestFramework, but the output is just:
Run tests:  <paths to DLLs>
  Failed! - Failed: XXX, Passed: YYY, Skipped: ZZZ, Total: VVVV, Duration: 1h 04m 54s 236ms
<full path>.dll : error run failed: Tests failed: '<full path to log file>.log' [net6.0|x64] 

How can I get the previous behavior of the console logger?
Thanks in advance!

**Comment by nohwnd** [comment, marked as answer]
Looks like your project is using the new testing platform and the new dotnet test integration that we are now working to improve. 

If you are using single project please try using `dotnet run` rather than `dotnet test`. 

You can also use: 

```
 dotnet run -- --output detailed
```
to see passed tests (like before with `--logger "console;verbosity=detailed"`)

**Reply by turbobobbytraykov** [reply]
That did the trick! Thanks!
I'm attempting to bring up to speed an old test automation project up to speed - at least in terms of underlying runner and overall test framework without any major changes.

I'm reading the docs over at https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli and I'm slowly getting to grips with the new Microsoft Test Platform.

**Reply by Evangelink** [reply]
I haven't done a test recently but I believe the same syntax should work for dotnet test.

About your automation project, if there's anything you would like to have as part of MSTest or the new platform, please let us know so we can investigate 🙏

**Reply by nohwnd** [reply]
> I haven't done a test recently but I believe the same syntax should work for dotnet test.

```
dotnet test --logger "console;verbosity=detailed"
```

This does work. 

The code below will NOT work, not before we will enable the new dotnet test experience for MSTest runner / testing.platform in net10.

```
dotnet test -- --output detailed
```

**Reply by turbobobbytraykov** [reply]
Thanks for tuning in @Evangelink!
Right now, I'm trying to limit my involvement in the solution/project to just tweaking the csproj file and Azure DevOps pipeline, hence I can change the invokation, so 'dotnet test' is not a mandatory requirement for me. I switched over to 'dotnet run' and made the necessary adjustments. Right now I'm investigating if filtering (based on TestCategory) is picking up thinks correctly.

Realistically, what I need to know is how much the Test Platform (and test runner, if applicable) can work with .net framework 4.X.
Our use case is supposed to target 4.6.2, 6.0 and 8.0 so I'm currently trying to see if 4.6.2 will hold up without any code changes.

Compatibility or incompatibility with certain libs/frameworks is what I'd really like to see stated clearly in the docs - this is my current request.

If I notice anything specific, I'll open a new discussion or issue about it.

**Reply by nohwnd** [reply]
Yes, the lib targets .NET Standard 2.0 (lowest supported target for that is .NET Framework 4.6.2), and .NET 6.0. So all the frameworks that are still in official support are also supported by Testing.Platform.


Good point about docs, I don't see any mention of compatibility with all supported frameworks there (either in testing.platform, or MSTest runner docs).
